### PR TITLE
EEG reference plugin

### PR DESCRIPTION
**Pull request by ViktorKL**
This pull request includes the EEGRef class in the utils lib and the EEG reference plugin for MNE-Scan

### CODE REVIEW

File: applications/mne_scan/plugins/reference/FormFiles/referencesetupwidget.h [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    ReferenceSetupWidget(Reference* pRef, QWidget *parent = 0);
+
+    //=========================================================================================================
+    /**
+    * Destroys the ReferenceSetupWidget.
+    * All ReferenceSetupWidget's children are deleted first. The application exits if ReferenceSetupWidget is the main widget.
+    */
+    ~ReferenceSetupWidget();
+
+
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/FormFiles/referencesetupwidget.h]: remove line

File: applications/mne_scan/plugins/reference/FormFiles/referencesetupwidget.h [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+private slots:
+    //=========================================================================================================
+    /**
+    * Shows the About Dialog
+    *
+    */
+    void showAboutDialog();
+
+private:
+
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/FormFiles/referencesetupwidget.h]: remove line

File: applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h [hunk 3]
```diff
@@ -0,0 +1,118 @@
+//=============================================================================================================
+/**
+* @file     reference.h
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h]: rename to referencetoolbarwidget.h

File: applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+* WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+* PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+* INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+* PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+* NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+*
+* @brief    Contains the declaration of the DummyYourWidget class.
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h]: REname to ReferenceToolbarWidget

File: applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+{
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// FORWARD DECLARATIONS
+//=============================================================================================================
+
+class Reference;
+
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h]: add line

File: applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+*
+* @brief The ReferenceToolbarWidget class provides a dummy toolbar widget structure.
+*/
+
+class REFERENCESHARED_EXPORT ReferenceToolbarWidget : public QWidget
+{
+    Q_OBJECT
+
+public:
+    typedef QSharedPointer<ReferenceToolbarWidget> SPtr;         /**< Shared pointer type for DummyYourWidget. */
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h]: Rename docu to ReferenceToolbarWidget

File: applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    typedef QSharedPointer<ReferenceToolbarWidget> ConstSPtr;    /**< Const shared pointer type for DummyYourWidget. */
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/FormFiles/referencetoolbarwidget.h]: see above

File: applications/mne_scan/plugins/reference/reference.cpp [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+
+    while(m_bIsRunning)
+    {
+        //Dispatch the inputs
+        MatrixXd t_mat = m_pRefBuffer->pop();
+
+        // apply common average reference
+        MatrixXd matCAR = EEGRef::applyCAR(t_mat, m_pFiffInfo);
+
+//        // write in- and output matrix to a file
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/reference.cpp]: pleae remove following commented code if not needed anymore

File: applications/mne_scan/plugins/reference/reference.h [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    */
+    void update(SCMEASLIB::NewMeasurement::SPtr pMeasurement);
+
+protected:
+    //=========================================================================================================
+    /**
+    * IAlgorithm function
+    */
+    virtual void run();
+
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/reference.h]: remove line

File: applications/mne_scan/plugins/reference/reference.pro.user [hunk 10]
```diff
@@ -0,0 +1,315 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (LorenzE) [inline applications/mne_scan/plugins/reference/reference.pro.user]: pleae remove this whole file from your PR

File: applications/mne_scan/plugins/ssvepbci/ssvepbci.cpp [hunk 11]
```diff
@@ -101,10 +101,10 @@ SsvepBci::SsvepBci()
 
 
     // Intitalise BCI data
-    //m_slChosenChannelsSensor << "9Z" << "8Z" << "7Z" << "6Z" << "9L" << "8L" << "9R" << "8R"; //<< "TEST";
-    m_slChosenChannelsSensor << "24" << "25" << "26" << "28" << "29" << "30" << "31" << "32";
-    //m_lElectrodeNumbers << 33 << 34 << 35 << 36 << 40 << 41 << 42 << 43;
-    m_lElectrodeNumbers << 24 << 25 << 26 << 28 << 29 << 30 << 31 << 32;
+    m_slChosenChannelsSensor << "9Z" << "8Z" << "7Z" << "6Z" << "9L" << "8L" << "9R" << "8R"; //<< "TEST";
```
Comment (LorenzE) [inline applications/mne_scan/plugins/ssvepbci/ssvepbci.cpp]: Is this correct setup or just for testing?

File: MNE/utils/eegref.h [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+    /**
+    * Constructs a EEGRef object.
+    */
+    EEGRef();
+
+    //=========================================================================================================
+    /**
+    * transforms the EEG data matrix with indifferent electrode reference to an EEG data matrix with common average reference. Bad channels are set to zero.
+    *
+    * @param[in] matIER         EEG data matrix with indefferent electrode reference
```
Comment (LorenzE) [inline MNE/utils/eegref.h]: indifferent

File: applications/mne_scan/plugins/ssvepbci/ssvepbci.cpp (hunk 11 above)
Comment (ViktorKL) [inline applications/mne_scan/plugins/ssvepbci/ssvepbci.cpp]: It's from our cap at BMTI which was tested with my BCI

**Pull request merged**